packages/system/src/use-style-config.ts

```diff
@@ -72,9 +80,78 @@ export function useStyleConfig(themeKey: any, props: any = {}, opts: any = {}) {
     }
 
     return stylesRef.current
-  }, [styleConfig, mergedProps, opts?.isMultiPart])
+  }, [breakpoints, styleConfig, mergedProps, opts?.isMultiPart])
 }
 
 export function useMultiStyleConfig(themeKey: string, props: any) {
   return useStyleConfig(themeKey, props, { isMultiPart: true })
 }
+
+interface ResolveResponsivePropStylesOptions {
+  responsiveValue: ResponsiveValue<string>
+  responsiveStyles: Record<string, SystemStyleObject>
+  breakpoints: string[]
+  props: Record<string, any>
+}
+
+function resolveResponsivePropStyles({
+  responsiveValue,
+  responsiveStyles,
+  breakpoints,
+  props,
+}: ResolveResponsivePropStylesOptions) {
+  if (responsiveValue == null) {
+    return {}
+  }
+
+  // for example: ["base", "sm", "md", "lg", "xl", "2xl"]
+  const breakpointKeys = Object.keys(breakpoints).filter(isCustomBreakpoint)
+
+  // If `responsiveValue` is a primitive value (not responsive),
+  // wrap the value into an array.
+  const sanitizedValue =
+    typeof responsiveValue === "string" || typeof responsiveValue === "number"
+      ? [responsiveValue]
+      : responsiveValue
```

Comment:
Could we do an early return here for the non responsive values to skip the computational overhead below?